### PR TITLE
Add metadata setter and getter to DU client

### PR DESCRIPTION
**Pull request by jtakalai**
also new member stipend because why not. TODO: document all of it...

### CODE REVIEW

File: packages/client/src/DataUnion.ts [hunk 1]
```diff
@@ -487,18 +521,27 @@ export class DataUnion {
...
+
+    /**
+     * Stores a Javascript object in JSON format in the data union contract, can be retrieved with `getMetadata`
+     * @param metadata object to be stored in the data union contract
+     */
+    async setMetadata(metadata: object): Promise<ContractReceipt> {
+        return this.sendAdminTx(this.contract.setMetadata, JSON.stringify(metadata))
+    }
+
+    async setMetadataString(metadata: string): Promise<ContractReceipt> {
```
Comment (hpihkala) [inline packages/client/src/DataUnion.ts:535]: Probably not necessary / good to expose a method to the end-user that enables them to break things. Better to enforce writing JSON to the metadata field.

**Review by hpihkala — CHANGES_REQUESTED** [review]
Otherwise LGTM

File: packages/client/src/DataUnion.ts (hunk 1 above)
Comment (jtakalai) [reply packages/client/src/DataUnion.ts:536]: ok so just remove the those "raw string" methods?

Comment (jtakalai) [reply packages/client/src/DataUnion.ts:536]: I was thinking it could be a URL or something, but of course it could just as well be wrapped in JSON  🤷

Comment (hpihkala) [reply packages/client/src/DataUnion.ts]: > ok so just remove the those "raw string" methods?

Yeah I would say so

**Pull request merged**